teps/0152-tekton-conformance-verification.md

```diff
@@ -0,0 +1,166 @@
+
+Tekton OSS repository will house the versioned conformance test suite. This suite will provide input Tekton Primitive YAMLs for the vendor service and then verify the corresponding outputs. Tekton vendors will be able to pull the test suite from the OSS repository and integrate it with their own implementations of the required `ProcessAndSendToTekton` function. However, the internal workings of their services will remain a "black box" to the test suite, as it will only evaluate the outputs.
+
+
+![conformance verification](images/0152-conformance-verification.png)
+
+
+
+More specifically for the verification process:
+- The test suite provides Tekton Primitives YAMLs as inputs for the vendor service to process. The YAMLs are vanilla Tekton-native PipelineRuns and TaskRuns that are vendor-agnostic. Each field/feature will be isolated in separate tests as much as possible to simplify failure attribution.
+
+- The black box testing of the vendor focuses solely on the outputs of the service, disregarding its internal implementation details. The vendor service takes ownership of monitoring CRD reconciliation and delivering compatible Tekton YAML outputs.
+
+- For verifying results, upon receiving [Tekton client](https://github.com/tektoncd/pipeline/blob/main/test/clients.go#L56-L71) readable YAML outputs,  the test suite will leverage the parse pkg for validating whether all the required conformant API fields are populated and the behaviors are honored.
+
+Tekton vendors will have the responsibility of integrating the conformance verification test suite with their respective services. To deliver conformance validation results to the test suite, vendors must implement the ProcessAndSendToTekton function. This function transforms vanilla Tekton Pipeline YAMLs into Tekton-parsable YAMLs for results verification.
+
+```go
+// ProcessAndSendToTekton accepts vanilla Tekton PipelineRun and TaskRun YAMLs, waits for the object to complete and outputs the completed PipelineRun and TaskRun along with the status.
+
+// The function parameters are the inputYAML and its corresponding Primitive type (either {PipelineRun} or {TaskRun}). Additionally, the function allows for the inclusion of custom inputs as variadic parameters.
+
+// The function will return the completed output YAML string and any associated errors.
+func ProcessAndSendToTekton(inputYAML, type string, customInputs …interface{}) (string, error)
+```
+
+#### Alternatives Considered
+As an alternative, Tekton vendors could expose endpoints for Tekton to call, allowing for  consumption and output of results according to HTTP API conformance (out-of-scope for the existing Tekton Conformance policy). In this scenario, Tekton Pipeline could trigger the testing process autonomously.
```

Comment:
Another con is that it adds more work (and cost) on tekton oss. Vendors should handle those costs if they aim to be conformant.

Reply:
Added, thanks Vincent.